client/src/app/test-list/test-list.component.ts

```diff
@@ -47,7 +48,12 @@ export class TestListComponent implements OnInit {
     this.dataSourceDetailed.paginator = this.paginator;
     this.dataSourceDetailed.sort = this.sort;
 
-    this.showDetailedTable = true;
```

Comment:
Tutaj powinno zostać `this.showDetailedTable = true;` zamiast tego if-elsa, którego dodałeś. Zrobiłem to dosyć prosto i nie zakładałem opcji chowania tej tabelki na dole. Jeśli dodamy if-elsa, to wtedy kliknięcie `Choose` przy innym teście spowoduje schowanie tabelki zamiast jej wyświetlenie.